### ISSUE TITLE
Coinbase create_order postOnly = True not reflected in the returned order ?

### ISSUE THREAD

**Issue by jtashiro** [ccxt/ccxt#23200]
### Operating System

ubuntu and mac

### Programming Languages

Python

### CCXT Version

4.3.66

### Description

When creating an limit order with params for postOnly = True, I expect the returned order object to reflect postOnly: True, however, I am seeing 'None'.   Is this a bug ?

### Code

```
command:
order = exchange.create_order(BTC/USDC, type=limit, side=buy, amount=7.536199703829913e-05, price=66346.44), params={'postOnly': True, 'timeInForce': 'PO'}

output:
[2024-07-24 08:10:21] {'amount': None,
 'average': None,
 'clientOrderId': '[removed]',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '[removed]',
 'info': {'client_order_id': '[removed],
          'order_id': '[removed]',
          'product_id': 'BTC-USDC',
          'side': 'BUY'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 **'postOnly': None,**
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'BTC/USDC',
 'takeProfitPrice': None,
 **'timeInForce': None,**
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}

```

**Comment by sc0Vu** [comment]
Hi @jtashiro,

I didn't see postOnly in the response of `createOrder`. @carlosmiei should we infer this value from the users parameter?

**Comment by jtashiro** [comment]
From what I can see on Coinbase site, the parameter is respected and the limit orders that execute have been maker only.  

Note that the majority of response variables are 'None' - is that something to look into in general?  I would expect at a minimum 'type' = limit and amount = <whatever I input> and price=<limit price> in the response.

**Comment by carlosmiei** [comment]
> From what I can see on Coinbase site, the parameter is respected and the limit orders that execute have been maker only.
> 
> Note that the majority of response variables are 'None' - is that something to look into in general? I would expect at a minimum 'type' = limit and amount = and price= to reflect user arguments, in the response.

@jtashiro Usually we avoid populating the order with inferred fields or the fields provided by the user because for some reason those can be slightly different after the execution. 

Can you see those fields (like postOnly) after doing a `fetch_order`?

**Comment by jtashiro** [comment]
Yes, the response from fetch_order on an Open order is well populated.  Not sure what the lag may be to call fetch_order after create_order.

          'order_configuration': {'limit_limit_gtd': {'base_size': '0.00049236',
                                                      'end_time': '2024-07-26T10:57:04.863256Z',
                                                      'limit_price': '60930.51',
                                                      'post_only': True}},

Having said that, when a limit order is submitted and accepted the response is immediate and I would not expect it to reflect the execution values, as it is likely still open.

can ccxt library have a parameter to have create_order call fetch_order after order is submitted/accepted, to return values from the exchange?   That would make this backward compatible and you wouldn't have to infer any values from user input.

**Comment by carlotestor** [comment]
Not a bug — expected. Coinbase's `createOrder` response only echoes back a handful of fields (`order_id`, `client_order_id`, `product_id`, `side`), so CCXT has nothing to parse the rest from. As a rule CCXT does not backfill order fields from your own input, because the exchange can silently adjust them (precision rounding, TIF downgrades, partial rejection).

The order *is* post-only — you confirmed it on the Coinbase UI. To get a fully populated structure, do a follow-up fetch:

```python
order = exchange.create_order('BTC/USDC', 'limit', 'buy', 0.00007536, 66346.44,
                              {'postOnly': True})
order = exchange.fetch_order(order['id'], 'BTC/USDC')
print(order['postOnly'], order['timeInForce'], order['amount'], order['price'])
```

There's no lag to worry about — the order exists as soon as `create_order` returns, so `fetch_order` immediately after is safe. If you need this constantly, `watch_orders()` (ccxt.pro) pushes the fully parsed order on every state change without the extra REST round-trip.
